internals/overlord/logstate/gatherer.go

```diff
@@ -132,18 +138,31 @@ func fillDefaultOptions(options *logGathererOptions) *logGathererOptions {
 
 // PlanChanged is called by the LogManager when the plan is changed, if this
 // gatherer's target exists in the new plan.
-func (g *logGatherer) PlanChanged(pl *plan.Plan, buffers map[string]*servicelog.RingBuffer) {
+func (g *logGatherer) PlanChanged(pl *plan.Plan, serviceData map[string]*ServiceData, labels map[string]map[string]string) {
```

Comment:
Update the comment above to say that PlanChanged expects labels[serviceName] value to never change after it is passed in. Otherwise we need to copy it before we send it on the channel.